main.tf

```diff
@@ -34,10 +34,13 @@ resource "azurerm_kubernetes_cluster" "main" {
     client_secret = var.client_secret
   }
 
-  addon_profile {
-    oms_agent {
-      enabled                    = true
-      log_analytics_workspace_id = azurerm_log_analytics_workspace.main.id
+  dynamic addon_profile {
+    for_each = var.enable_log_analytics_workspace ? ["log_analytics"] : []
+    content {
```

Comment:
I may miss some tech background but do we need this "content{}" block? Can we just have the "ome_agent{}" within the "addon_profile{}" w/o the "content{}"?

Reply:
No. Content is required in dynamic blocks.